screens/Challenges/ToDoScreen.js

```diff
@@ -1,48 +1,119 @@
+          challenge={challenge}
+          mainColor={alternator ? Colors.KIT_LIGHT_ORANGE : Colors.KIT_GREEN}
+          onPress={() => {
+            alert("Hello");
+          }}
+        />
+      );
       alternator = !alternator;
     }
-    return todos
+    return todos;
   }
 
   useEffect(() => {
-    currentUser = getUserId();
+    const currentUser = getUserId();
     getAllAssignedChallenges(currentUser).then(challenges => {
-      setChallenges(challenges)
-    })
+      const openedChallenges = [];
+      const unopenedChallenges = [];
+      // Filter opened and unopened challenges
+      for (const challenge of challenges) {
+        let opened = false;
+        if (challenge.hasOwnProperty("opened")) {
+          const openedUserObjects = challenge.opened;
+          for (const openedUser of Object.keys(openedUserObjects)) {
+            if (openedUser === currentUser) {
+              opened = true;
+              break;
+            }
+          }
+        }
+        if (opened) {
+          openedChallenges.push(challenge);
+        } else {
+          unopenedChallenges.push(challenge);
+        }
+      }
+      setOpenedChallenges(openedChallenges);
+      setUnopenedChallenges(unopenedChallenges);
+      setChallengeCards(unopenedChallenges);
+    });
   }, []);
 
+  const getTimeLeftInMilliseconds = assignedTime => {
+    // 24 hours in seconds
+    const TIME_TO_OPEN = 86400000;
+    const currTime = moment().valueOf(); // Gets unix time in milliseconds
+    // console.log(moment(assignedTime).format("HH:MM:SS"));
+    // console.log(moment(currTime).format("HH:MM:SS"));
+    // console.log(assignedTime + TIME_TO_OPEN - currTime);
+    return assignedTime + TIME_TO_OPEN - currTime;
+  };
+
   return (
     <ScrollView>
       <View style={styles.container}>
-        <ChallengesScreen />
-        <KitText style={styles.label} fontWeight="medium" size={24}>
-          To Do
-        </KitText>
-        {challenges.length > 0 && renderTodos()}
+        {challengeCards.length !== 0 && (
+          <View style={styles.cardContainer}>
+            {challengeCards.map((challenge, index) => (
+              <View key={index} style={styles.card}>
+                <OpenChallengeCard
+                  assignedChallengeId={challenge.assignedChallengeId}
+                  numberOfChallenges={challengeCards.length}
+                  challengeTitle={challenge.challengeDetails.title}
+                  challengeDescription={challenge.challengeDetails.description}
+                  teamName={challenge.teamName}
+                  timeLeftInMilliseconds={getTimeLeftInMilliseconds(
+                    challenge.assignedTime
+                  )}
+                  onReveal={() => {
+                    openChallenge(challenge.assignedChallengeId, getUserId());
+                  }}
+                  onClose={() => {
+                    setChallengeCards(
+                      challengeCards.filter((card, i) => i !== index)
```

Comment:
I'm not 100% on what this does, but I assume it removes the challenge card that triggered the `onClose` from the list of unrevealed challenges and adds it to the list of revealed challenges (via hooks).

A comment might be nice

Reply:
It removes the challenge card from the list of challengeCards but does not add it to the list of opened challenges (this should be done by doing a call to firebase, yet to be implemented). The reason I chose to separate challengeCards from openedChallenges and unopenedChallenges is because the card should still be shown even if the challenge has been opened.